x/bonds/internal/types/types.go

```diff
@@ -30,43 +31,43 @@ func CheckReserveTokenNames(resTokens []string, token string) sdk.Error {
 	return nil
 }
 
-func CheckNoOfReserveTokens(resTokens []string, fnType string) sdk.Error {
+func CheckNoOfReserveTokens(resTokens []string, fnType string) error {
 	// Come up with number of expected reserve tokens
 	expectedNoOfTokens, ok := NoOfReserveTokensForFunctionType[fnType]
 	if !ok {
-		return ErrUnrecognizedFunctionType(DefaultCodespace)
+		return sdkerrors.Wrap(ErrUnrecognizedFunctionType, DefaultCodespace)
```

Comment:
I think instead of `DefaultCodespace` we should have `fnType`.

Reply:
Great catch